src/ipv4.c

```diff
@@ -527,7 +527,7 @@ int ipv4_xmit(struct rte_mbuf *mbuf, const struct flow4 *fl4)
     struct route_entry *rt;
     struct ipv4_hdr *iph;
 
-    if (!mbuf || !fl4 || fl4->fl4_saddr.s_addr == htonl(INADDR_ANY)) {
+    if (!mbuf || !fl4) {
```

Comment:
Why  remove source check? Does "traceroute" work after removing it?
I think the following aspects should be considered for traceroute:
1. traceroute should support "virtual ip or host ip" configured in dpvs, some icmp types, such as ICMP_DEST_EXCEEDED, ICMP_DEST_UNREACH, may need to construct.
2. traceroute shoud support ipvs snat mode, dpvs should support the forwarding of traceroute udp/icmp packets.
3. traceroute should support both ipv4 and ipv6.
2. traceroute should

Reply:
> Why remove source check? 

maybe the question is why check source address here?  in this function,  line 654   `if (iph->src_addr == htonl(INADDR_ANY)) {` is checking src_addr eq ANY, and will select an address on the port as src_addr.
this function is called only at `icmp_echo` and `icmp_send`.
`icmp_echo` is for `ICMP_ECHO` processing. it swaped `src_addr` & `dst_addr`, then called `ipv4_xmit`. this change does nothing different.
`icmp_send` is used for generating icmp packets. if `dst_addr` is local address, the out icmp packet will use it as source address. if not, should select an local address for the out packet. just set `src_addr` `ANY`, and `ipv4_xmit` will do the selection. 
the selection first get the out interface by `route4_output`, and then use `inet_addr_select` to get a local address on that port.
@ywc689

Reply:
> > Why remove source check?
> 
> maybe the question is why check source address here? in this function, line 654 `if (iph->src_addr == htonl(INADDR_ANY)) {` is checking src_addr eq ANY, and will select an address on the port as src_addr.
> this function is called only at `icmp_echo` and `icmp_send`.
> `icmp_echo` is for `ICMP_ECHO` processing. it swaped `src_addr` & `dst_addr`, then called `ipv4_xmit`. this change does nothing different.
> `icmp_send` is used for generating icmp packets. if `dst_addr` is local address, the out icmp packet will use it as source address. if not, should select an local address for the out packet. just set `src_addr` `ANY`, and `ipv4_xmit` will do the selection.
> the selection first get the out interface by `route4_output`, and then use `inet_addr_select` to get a local address on that port.
> @ywc689

this is used for ipvs snat mode.